addons/im_livechat/models/discuss_channel.py

```diff
@@ -196,7 +186,8 @@ def _chatbot_find_customer_values_in_messages(self, step_type_to_field):
         """
         values = {}
         filtered_message_ids = self.chatbot_message_ids.filtered(
-            lambda m: m.script_step_id.step_type in step_type_to_field.keys()
+            # sudo: chatbot.script.step - getting the type of the current step
+            lambda m: m.script_step_id.sudo().step_type in step_type_to_field
```

Comment:
I'm surprised that the step itself is accessible

Reply:
I'm surprised as well, I found out that adding `.sudo(False)` to `current_step` (chatbot.py#46) prevent from accessing the step itself. This method is only called from this flow.